src/select/parts/item.tsx

```diff
@@ -33,10 +36,56 @@ export interface ItemProps {
   highlightType?: HighlightType['type'];
   withScrollbar?: boolean;
   sticky?: boolean;
+  renderOption?: T;
+  parentProps?: ItemParentProps;
 }
 
+export interface ItemParentProps {
+  index: number;
+  virtualIndex?: number;
+  option: DropdownOption;
+  disabled: boolean;
+}
+const toSelectOptionGroupItem = (props: ItemParentProps): SelectProps.SelectOptionGroupItem => {
+  return {
+    type: 'group',
+    index: props.virtualIndex ?? props.index,
+    option: props.option.option as OptionGroup,
+    disabled: props.disabled,
+  };
+};
+
+const toSelectOptionItem = (props: {
+  index: number;
+  virtualIndex?: number;
+  option: DropdownOption;
+  disabled: boolean;
+  selected: boolean;
+  highlighted: boolean;
+  parentProps?: ItemParentProps;
+}): SelectProps.SelectOptionItem => {
+  return {
+    type: 'item',
+    index: props.virtualIndex ?? props.index,
+    option: props.option.option as OptionDefinition,
+    selected: props.selected,
+    highlighted: props.highlighted,
+    disabled: props.disabled,
+    parent: props.parentProps
+      ? toSelectOptionGroupItem({
```

Comment:
Since this creates a new object, is there a chance that this creates issues for customers, for example if they try to do determine parent-child relationships by performing identity comparisons?